hermes/bin/src/runtime_extensions/hermes/sqlite/core.rs

```diff
@@ -0,0 +1,198 @@
+/// ! Core functionality implementation for the `SQLite` open function.
+use libsqlite3_sys::{
+    sqlite3, sqlite3_exec, sqlite3_open_v2, sqlite3_soft_heap_limit64, SQLITE_ERROR, SQLITE_OK,
+    SQLITE_OPEN_CREATE, SQLITE_OPEN_READONLY, SQLITE_OPEN_READWRITE,
+};
+
+use crate::{
+    app::HermesAppName,
+    runtime_extensions::{
+        app_config::{get_app_in_memory_sqlite_db_cfg, get_app_persistent_sqlite_db_cfg},
+        bindings::hermes::sqlite::api::Errno,
+    },
+};
+
+/// Represents the various errors that can occur when opening a database.
+#[derive(Debug)]
+pub(super) enum OpenError {
+    /// The in-memory configuration provided is invalid.
+    InvalidInMemoryConfig,
+    /// The persistent configuration provided is invalid.
+    InvalidPersistentConfig,
+    /// The database name is missing in the persistent configuration.
+    MissingDatabaseNameForPersistentConfig,
+    /// Failed to open the database.
+    FailedOpeningDatabase,
+    /// Failed to set the database size.
+    FailedSettingDatabaseSize,
+    /// An error occurred with `SQLite`, represented by an `Errno`.
+    SQLiteError(Errno),
+}
+
+/// Opens a connection to a new or existing `SQLite` database.
+pub(super) fn open(
```

Comment:
I am wondering, why not to make this `open` connection function as a part of `connection` module ?

Reply:
Because the constructor in WIT does return only the object itself. In this case, `open` can possibly causes an error that's why we need a separate function to handle with `Result<object>`.